msal4j-brokers/src/test/java/test/ProofOfPossessionTest.java

```diff
@@ -0,0 +1,192 @@
+package test;
+
+import com.microsoft.aad.msal4j.*;
+import com.microsoft.aad.msal4jbrokers.MsalRuntimeBroker;
+import infrastructure.SeleniumExtensions;
+import labapi.LabUserProvider;
+import labapi.User;
+import org.openqa.selenium.WebDriver;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+@Test()
+public class ProofOfPossessionTest {
+    private final static String MICROSOFT_AUTHORITY_ORGANIZATIONS =
+            "https://login.microsoftonline.com/organizations/";
+    private final static String GRAPH_DEFAULT_SCOPE = "https://graph.windows.net/.default";
+
+    private LabUserProvider labUserProvider;
+
+    WebDriver seleniumDriver;
+
+    @BeforeClass
+    public void setUp() {
+        labUserProvider = LabUserProvider.getInstance();
+    }
+
+    @Test
+    public void acquirePopToken_WithBroker() throws Exception {
+        User user = labUserProvider.getDefaultUser();
+
+        MsalRuntimeBroker broker = new MsalRuntimeBroker();
+
+        PublicClientApplication pca = createPublicClientApp(user, broker);
+
+        IAuthenticationResult result = acquirePoPTokenUsernamePassword(pca, user, Collections.singleton(GRAPH_DEFAULT_SCOPE));
+
+        //A valid PoP access token should be returned if a broker was set
+        assertTokenResultNotNull(result);
+        Assert.assertTrue(result.isPopAuthorization());
+    }
+
+    @Test(expectedExceptions = MsalClientException.class)
+    public void acquirePopToken_WithoutBroker() throws Exception {
+        User user = labUserProvider.getDefaultUser();
+
+        PublicClientApplication pca = createPublicClientApp(user);
+
+        //Setting UserNamePasswordParameters.proofOfPossession without enabling the broker should result in an exception when trying to get a token
+        IAuthenticationResult result = acquirePoPTokenUsernamePassword(pca, user, Collections.singleton(GRAPH_DEFAULT_SCOPE));
+    }
+
+    @Test
+    public void acquirePopToken_BrowserAndBroker() throws Exception {
+        User user = labUserProvider.getDefaultUser();
+
+        seleniumDriver = SeleniumExtensions.createDefaultWebDriver();
+
+        //First, get a non-PoP (bearer) token through a browser
```

Comment:
Why not use non-WAM Username/Password? UI testing is brittle, should be avoided as much as possible.